cow_disease_detection/fetch_data/pull_and_clean_data.py

```diff
@@ -80,6 +80,22 @@ def get_data() -> pd.DataFrame:
     sh = gc.open_by_key(key)
     worksheet = sh.sheet1
     df = get_as_dataframe(worksheet)
+
+    # To pull historical data from a specific date
+    args = argument_parser()
+
+    # change datatypes
+    date = pd.to_datetime(df["date"])
+    date_2 = pd.to_datetime(args.pull_data_from_date)
+
+    # using a loop to get the first index value of the given date.
+    for index, value in enumerate(date):
+        if value == date_2:
+            df = df.iloc[
+                index:,
+            ]
+            break
```

Comment:
1. Always create your branch name that shows what you have done, e.g. here, the branch name should have been: feature/new_deate_argument. if it was a bug fix then the name should have been bugfix/data_time_bug etc
2. I do not understand the logic/ flow of your solution. My solution would be:
    2a. get_data function takes a new argument 
    2b. convert the new argument to date format
    2c. Filter/Index the data using the argument in 2b 
That's it !

Also, you can change the flow a little bit and bring the "date_time" column calculations from the data_preprocessing function to get_data function. This will make it easier to handle and we won't be parsing columns for dates at different functions